src/cryptonote_core/service_node_quorum_cop.cpp

```diff
@@ -249,8 +249,22 @@ namespace service_nodes
 
               auto test_results = check_service_node(node_key, info);
 
+              bool passed = test_results.uptime_proved && test_results.voted_in_checkpoints;
+
+              if (!test_results.voted_in_checkpoints &&
+                  m_core.get_nettype() == cryptonote::MAINNET &&
+                  m_obligations_height < HF_VERSION_12_CHECKPOINTING_SOFT_FORK_HEIGHT)
```

Comment:
This `if` should also have a `test_results.uptime_proved` check (so that it doesn't change a fail to a pass if it's missing uptime proofs *and* checkpoint votes).

Reply:
Fixed